proposals/0475-tuple-syntax.rst

```diff
@@ -63,15 +63,27 @@ variable in a bit of code. This allows us to e.g. refer to ``Tuple2`` and ``Tupl
 Proposed Change Specification
 -----------------------------
 
-1. Create a new module in ``base`` called ``GHC.Prelude``. This module exports definitions
-   peculiar to GHC that are safe to use in ordinary code.
-   This module is *not* intended as a replacement to ``Prelude``,
-   but instead a complement to it. This module is a counterpart to ``GHC.Exts``, which exports
+1. Create three new modules in ``ghc-experimental`` called ``GHC.Prelude``,
```

Comment:
`GHC.Prelude` is actually taken by the GHC API for the Prelude used by GHC itself. Are we absolutely sure we want to repurpose this name, when we could simply pick a different one? Need opinion from @goldfirere

Reply:
Huh -- that is annoying.

It demonstrates that we are using the module prefix "GHC." in two different ways:
* Modules of GHC itself (e.g. GHC.Core.Opt.Simplify)
* Modules in `ghc-experimental`.
These are not the same purpose.

Maybe `ghc-experimental` should start with `GHCX.`?  Thus `GHCX.Prelude`, `GHCX.Tuple`.   Or some other prefix?

Reply:
`GHCX.*` is just a slightly different take on `GHC.Exts.*` that @tek previously suggested, right? I believe `GHC.Exts.*` is good and descriptive. For example, `GHC.Exts.Tuple` is "GHC's extensions to tuples". When those get stabilized, they can be reexported from `Data.Tuple`

Reply:
I wonder about the broader situation here - would the idea be that all `GHC.Exts.*` modules end in non-`GHC.Exts.*` modules if/when brought into `base`? Since if anything is added to `base` *also* under the `GHC.Exts.*` moniker, you would still end up with a naming conflict during the deprecation period from `ghc-experimental`

Reply:
`GHC.Exts` exposes some unsafe operations, as in "crash the process" unsafe, which the original text uses as contrast for the purpose of `GHC.Prelude`…would calling the new namespace `GHC.Exts.*` associate `ghc-experimental` with that notion?

On the other hand, `GHC.Exts` might be a candidate for migration to `ghc-internals`.

There is also a large number of `GHC.*` modules in `base` that might be a source for additional conflicts later.

Reply:
I find the prefix `GHC.` for things in ghc-experimental strange, as they are neither about GHC (e.g. a plugins) or part of GHC (e.g. GHC.Prelude). It's as if I start to prefix my library modules with `Nomeata.`. I expect developers to find this naming confusing (me already included.)

How about we append `Experimental` to the module name from base where things might end up in? `Data.Tuple.Experimental`?

Reply:
The only thing is should GHC's experiments have a monopoly on such names?

`Data.Tuple.GlasgowExperiment` lol

What it is comes first, "compilation" is not relevant and thus unmentioned, who experiment and so not to conflict with other experiments comes last.

Reply:
now called `Prelude.Experimental`.